NetCord/Rest/RestMessage.cs

```diff
@@ -75,67 +78,170 @@ public RestMessage(NetCord.JsonModels.JsonMessage jsonModel, RestClient client)
+    /// A dictionary of <see cref="GuildChannelMention"/> objects indexed by their IDs, containing channels specifically mentioned in the message.
+    /// </summary>
+    /// <remarks>
+    /// Not all channel mentions in a message will appear in <see cref="MentionedChannels"/>. Only <see cref="TextChannel"/>s visible to everyone in a lurkable guild will ever be included. Only crossposted messages (via Channel Following) currently include <see cref="MentionedChannels"/> at all. If no <see cref="Mention"/>s in the message meet these requirements, this field will not be sent.
+    /// </remarks>
     public IReadOnlyDictionary<ulong, GuildChannelMention> MentionedChannels { get; }
 
+    /// <summary>
+    /// A dictionary of <see cref="Attachment"/> objects indexed by their IDs, containing any files attached in the message.
+    /// </summary>
     public IReadOnlyDictionary<ulong, Attachment> Attachments { get; }
 
+    /// <summary>
+    /// A list of <see cref="Embed"/> objects containing any embedded content present in the message.
+    /// </summary>
     public IReadOnlyList<Embed> Embeds { get; }
 
+    /// <summary>
+    /// A list of <see cref="MessageReaction"/> objects containing all reactions to the message.
+    /// </summary>
     public IReadOnlyList<MessageReaction> Reactions { get; }
 
+    /// <summary>
+    /// Used for validating that a message was sent.
+    /// </summary>
     public string? Nonce => _jsonModel.Nonce;
 
+    /// <summary>
+    /// Whether this message is pinned in a channel.
+    /// </summary>
     public bool IsPinned => _jsonModel.IsPinned.GetValueOrDefault();
 
+    /// <summary>
+    /// If the message was generated by a <see cref="Webhook"/>, this is the webhook's ID.
+    /// </summary>
     public ulong? WebhookId => _jsonModel.WebhookId;
 
+    /// <summary>
+    /// The type of the message.
+    /// </summary>
     public MessageType Type => _jsonModel.Type.GetValueOrDefault();
 
+    /// <summary>
+    /// Sent with Rich Presence-related chat embeds.
+    /// </summary>
     public MessageActivity? Activity { get; }
 
+    /// <summary>
+    /// Sent with Rich Presence-related chat embeds.
+    /// </summary>
     public Application? Application { get; }
 
+    /// <summary>
+    /// If the message is an <see cref="IInteraction"/> or application-owned <see cref="Webhook"/>, this is the ID of the <see cref="NetCord.Application"/>.
+    /// </summary>
     public ulong? ApplicationId => _jsonModel.ApplicationId;
 
+    /// <summary>
+    /// Contains data showing the source of a crosspost, channel follow add, pin, or message reply.
+    /// </summary>
     public MessageReference? MessageReference { get; }
 
+    /// <summary>
+    /// A <see cref="MessageFlags"/> object indicating the message's applied flags.
+    /// </summary>
     public MessageFlags Flags => _jsonModel.Flags.GetValueOrDefault();
 
+    /// <summary>
+    /// The message associated with the <see cref="MessageReference"/>.
```

Comment:
MessageReference doesn't have anything in common with MessageReference aside of similar name I think

Reply:
I'm fairly certain it does. That's the ID, and this is the message itself.